server/autotest_server/testers/rust/rust_tester.py

```diff
@@ -0,0 +1,160 @@
+            i += 1
+
+    return items
+
+
+class RustTester(Tester):
+    def __init__(self, specs: TestSpecs, test_class: Type[RustTest] = RustTest) -> None:
+        super().__init__(specs, test_class)
+
+    def parse_test_events(self, items: list[dict]) -> list[RustTest]:
+        tests = []
+
+        for item in items:
+            # Ignore suite events.
+            if item["type"] != "test":
+                continue
+
+            event = item["event"]
+
+            finished_events = {"failed", "ok"}
+
+            if event not in finished_events:
+                continue
+
+            executable, test = parse_test_identifier(item["name"])
+
+            output = item["stdout"] if event == "failed" else ""
+
+            tests.append(self.test_class(self, executable, test, event == "ok", output))
+
+        return tests
+
+    # This should likely be moved to setup.py's create_environment.
+    def rust_env(self) -> dict:
+        # Hint to /bin/sh that it should look in $HOME/.cargo/bin.
+        # Despite .profile pointing to this directory, /bin/dash does not want to acknowledge it.
+        # There is conflicting information online as to whether-or-not .profile is respected by dash.
+        rust_home_path = os.path.join(Path.home(), ".cargo", "bin")
+
+        env = os.environ.copy()
+
+        env["PATH"] = rust_home_path + ":" + env["PATH"]
+
+        return env
+
+    def compile_rust_tests(self, directory: str) -> subprocess.CompletedProcess:
+        command = ["cargo", "nextest", "run", "--no-run", "--color", "never"]
+
+        env = self.rust_env()
+
+        return subprocess.run(command, cwd=directory, env=env, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    # We need a way to get machine-readable output from cargo test (why are we using nextest?).
+    # Here are the options I've explored:
+    #  - cargo-test by default outputs human-readable strings.
+    #    These strings might change in format over time, and I figure one tests stdout could bleed into another.
+    #  - cargo-test supports machine-readable strings (via `cargo test -- --format json`).
+    #    This option is nightly rust only, and enforcing nightly rust could cause other issues.
+    #  - cargo-nextest supports experimental machine-readable output.
+    #    While machine-readable output is experimental, it's based on yhr libtest standard that cargo test would use.
+    #    nextest should also interact very similarly to `cargo test`. It should be very simple to swap to cargo-test.
+    #    It's also reliable and only requires Rust 1.36 or earlier for running.
+    def run_rust_tests(self, directory: str, module: Optional[str]) -> subprocess.CompletedProcess:
+        command = ["cargo", "nextest", "run", "--no-fail-fast", "--message-format", "libtest-json", "--color", "never"]
+
+        # Prevent CLI options from being propagated.
+        if module is not None and "-" not in module:
+            command.append(module)
```

Comment:
@1whatleytay I see on [nextest's documentation](https://nexte.st/book/running.html#options-and-arguments) that you're treating `module` as "`[FILTERS]`". I wasn't able to figure out exactly how that part worked, but at the very least it seems to match more than just the module name? With the sample files you added on your MarkUs PR, using a value of `test_enqueue` does a filter based on test name.

But, I also see the `-E '...'` test filter expression option, which seems [clearly defined](https://nexte.st/book/filter-expressions.html). I wonder if this is a better target for user configuration, if it supports filtering both by package name (e.g. `package(test)`?) and test function name (e.g. `test(dequeue)`). It would require the instructor to understand this syntax, but I think that's okay?